### DISCUSSION TITLE
Does OpenPNM Support Triangular Throats?

### DISCUSSION THREAD

**Discussion by UnaisAshraf** [PMEAL/OpenPNM#2824] [Q&A]
Dear OpenPNM community,

I'm curious to know whether OpenPNM, includes support for triangular throat shapes. Can anyone confirm if this feature is available in OpenPNM, and if so, provide some insights or documentation on how to utilize it?

Thank you

**Comment by jgostick** [comment]
Sadly we've never had the need for this though it's common in rock physics.  You can write our own pore scale models then use openpnm to do the calculations and such.  This is how we envisioned people would use openpnm.  An example of how to write a pore scale model is [here](https://openpnm.org/examples/reference/architecture/pore_scale_models.html)

**Comment by likeduck** [comment]
You can try Sataoil's file format, which contains four basic PNM files, including the shape factor G, which can represent triangular pore throats.
